### PR TITLE
Refactor initialization, set up WS connection

### PR DESCRIPTION
**Pull request by rpiaggio**
I moved all initialization to React. This way we can show a loader as soon as possible while we initialize. Also, we use the available hooks.

Added to the initialization is establishing the WS connection to the server and processing the received events.

### CODE REVIEW

File: modules/web/client/src/main/scala/observe/ui/components/MainApp.scala [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    _.flatMap:
+      case WSFrame.Text(text, _) => fs2.Stream(decode[ClientEvent](text))
+      case _                     => fs2.Stream.empty
+
+  // TODO IN CRYSTAL: It would be nice to have a .useEffectWhenDepReady instead of doing .void in dependencies!
+
+  private val component =
+    ScalaFnComponent
+      .withHooks[Unit]
+      .useResourceOnMount: // Build AppContext
```
Comment (cquiroz) [inline modules/web/client/src/main/scala/observe/ui/components/MainApp.scala:127]: This is a good idea, should we replicate in explore?

File: modules/web/client/vite.config.js [hunk 2]
```diff
@@ -139,9 +139,10 @@ export default defineConfig(async ({ mode }) => {
       https: true,
       proxy: {
         '/api': {
-          target: 'http://localhost:7070',
+          target: 'http://127.0.0.1:7070',
```
Comment (cquiroz) [inline modules/web/client/vite.config.js:142]: maybe `local.lucuma.xyz` would work too?

**Review by cquiroz — APPROVED** [review]
This seems like a very. good idea

File: modules/web/client/src/main/scala/observe/ui/components/MainApp.scala (hunk 1 above)
Comment (rpiaggio) [reply modules/web/client/src/main/scala/observe/ui/components/MainApp.scala:127]: We could do in explore, yes

**Pull request merged**